Model/Api/CreateOrder.php

```diff
@@ -0,0 +1,694 @@
+        StockRegistryInterface $stockRegistry,
+        SessionHelper $sessionHelper
+    ) {
+        $this->hookHelper = $hookHelper;
+        $this->orderHelper = $orderHelper;
+        $this->logHelper = $logHelper;
+        $this->request = $request;
+        $this->bugsnag = $bugsnag;
+        $this->response = $response;
+        $this->configHelper = $configHelper;
+        $this->cartHelper = $cartHelper;
+        $this->url = $url;
+        $this->backendUrl = $backendUrl;
+        $this->stockRegistry = $stockRegistry;
+        $this->sessionHelper = $sessionHelper;
+    }
+
+    /**
+     * Pre-Auth hook: Create order.
+     *
+     * @api
+     *
+     * @param string $type - "order.create"
+     * @param array $order
+     * @param string $currency
+     *
+     * @return void
+     * @throws \Exception
+     */
+    public function execute(
+        $type = null,
+        $order = null,
+        $currency = null
+    ) {
+        try {
+            if ($type !== 'order.create') {
+                throw new BoltException(
+                    __('Invalid hook type!'),
+                    null,
+                    self::E_BOLT_GENERAL_ERROR
+                );
+            }
+
+            $payload = $this->request->getContent();
+
+            $this->validateHook();
+
+            if (empty($order)) {
+                throw new BoltException(
+                    __('Missing order data.'),
+                    null,
+                    self::E_BOLT_GENERAL_ERROR
+                );
+            }
+
+            $quoteId = $this->getQuoteIdFromPayloadOrder($order);
+            /** @var Quote $immutableQuote */
+            $immutableQuote = $this->loadQuoteData($quoteId);
+
+            // Convert to stdClass
+            $transaction = json_decode($payload);
+
+            /** @var Quote $quote */
+            $quote = $this->orderHelper->prepareQuote($immutableQuote, $transaction);
+            $this->validateQuoteData($quote, $transaction);
+
+            /** @var \Magento\Sales\Model\Order $createOrderData */
+            $createOrderData = $this->orderHelper->preAuthCreateOrder($quote, $transaction);
+            $orderReference = $this->getOrderReference($order);
+
+            $this->sendResponse(200, [
+                'status'    => 'success',
+                'message'   => 'Order create was successful',
+                'display_id' => $createOrderData->getIncrementId() . ' / ' . $quote->getId(),
+                'total'      => $this->cartHelper->getRoundAmount($createOrderData->getGrandTotal()),
+                'order_received_url' => $this->getReceivedUrl($immutableQuote),
+            ]);
+        } catch (\Magento\Framework\Webapi\Exception $e) {
+            $this->bugsnag->notifyException($e);
+            $this->sendResponse($e->getHttpCode(), [
+                'status' => 'failure',
+                'error'  => [[
+                    'code' => self::E_BOLT_GENERAL_ERROR,
```

Comment:
Is there a more descriptive error we can use here, from my observation it seems to be more than just a general error?

Reply:
We do not have special error code for webhook failes, as example. So we use general code for such fails.